runtime-sdk/src/modules/core/types.rs

```diff
@@ -41,6 +45,17 @@ pub struct CallDataPublicKeyQueryResponse {
     pub epoch: u64,
 }
 
+/// Response to the public key query.
+#[derive(Clone, Debug, Default, cbor::Encode, cbor::Decode)]
+pub struct KeyManagerPublicKeyQueryResponse {
+    /// ID of the public key which signs the call data public keys
+    pub key_pair_id: KeyPairId,
+    /// Runtime signing key which signs the call data public keys
+    pub public_key: PublicKey,
+    /// For reference, which runtime ID is this for?
```

Comment:
When querying the key manager for (long or short term) keys you need to specify the runtime id (e.g. Sapphire) so that the key manager node can verify if the enclave, which is making the call, is allowed to query keys for the given runtime.